csob/api_response.py

```diff
@@ -0,0 +1,74 @@
+from typing import Optional
+
+from requests import Response
+
+from csob.enums import ResultCode, PaymentStatus
+from csob.exceptions import SERVICE_RESULT_CODE_EXCEPTION_DICT
+
+
+class APIResponse:
+    api_response: Optional[Response]
+    is_verified: Optional[bool]
+    _parsed_data: Optional[dict]
+
+    def __init__(self, api_response: Optional[Response], parsed_data: Optional[dict], is_verified: Optional[bool],
+                 raise_exception=False):
+        self._parsed_data = parsed_data
+        self.api_response = api_response
+        if parsed_data is None and api_response is None:
+            raise ValueError('You have to specify at least `api_response`')
+        self.is_verified = is_verified
+
+        if self.is_verified:
+            if raise_exception and self.is_okay is False:
+                raise SERVICE_RESULT_CODE_EXCEPTION_DICT[self.result_code]
+
+    @property
+    def response_json(self) -> Optional[dict]:
+        if self.api_response is not None:
+            if self.http_status_code == 200:
+                return self.api_response.json()
+        else:
+            return self._parsed_data
+
+    @property
+    def http_status_code(self) -> Optional[int]:
+        if self.api_response is not None:
+            return self.api_response.status_code
```

Comment:
a co se stane, když else?

Reply:
@matllubos Tak je navratová hodnota `None`

Reply:
no to jsi měl pochopit jako rejp, že tohle je pylint chyba. Vždy musíš něco vrátit  a to i None i když je default

Reply:
nejsem úplně hloupý, vím že funkce/metoda vrací by default None. Už nějakou dobu programuju :dagger: :smile:

Reply:
`import this`

Reply:
Mě to přišlo "hezčí" bez toho :smile: 
```
>>import this
The Zen of Python, by Tim Peters

Beautiful is better than ugly.
Explicit is better than implicit.
```
Ale přidám.

Reply:
Ano máš to hned v té druhé větě
Explicit is better than implicit.